src/main/java/school/hei/haapi/service/AwardedCourseService.java

```diff
@@ -87,4 +90,46 @@ public List<AwardedCourse> getByCriteria(
         PageRequest.of(page.getValue() - 1, pageSize.getValue(), Sort.by(ASC, "name"));
     return awardedCourseDao.findByCriteria(teacherId, courseId, pageable);
   }
+
+  public List<AwardedCourse> getByTeacherId(
+      String teacherId, PageFromOne page, BoundedPageSize pageSize) {
+    Pageable pageable =
+        PageRequest.of(page.getValue() - 1, pageSize.getValue(), Sort.by(ASC, "course"));
+    User teacher = userService.findById(teacherId);
+    return awardedCourseRepository.findByMainTeacherId(teacher.getId(), pageable);
+  }
+
+  @Transactional
+  public List<AwardedCourse> createOrUpdateAwardedCoursesByTeacherId(
+      String teacherId, List<AwardedCourse> awardedCourses) {
+
+    List<AwardedCourse> updatedAwardedCourses = new ArrayList<>();
+
+    awardedCourses.forEach(
+        awardedCourse -> {
+          if (awardedCourse.getId() != null) {
+            AwardedCourse existingAwardedCourse = findById(awardedCourse.getId());
+
+            existingAwardedCourse.setCourse(awardedCourse.getCourse());
+            existingAwardedCourse.setGroup(awardedCourse.getGroup());
+            existingAwardedCourse.setMainTeacher(awardedCourse.getMainTeacher());
+
+            awardedCourseValidator.accept(existingAwardedCourse);
+            updatedAwardedCourses.add(existingAwardedCourse);
+          } else {
+            awardedCourseValidator.accept(awardedCourse);
+            updatedAwardedCourses.add(awardedCourse);
+          }
+        });
+
```

Comment:
You not need to assert if the Id of the AwardedCourse is null because the database will create it automatically, just save instead

Reply:
and remove this if else condition